src/framework/auth/services/token/token.ts

```diff
@@ -0,0 +1,102 @@
+import { urlBase64Decode } from '../../helpers';
+
+export interface NbAuthToken {
+  getValue(): string;
+  isValid(): boolean;
+  toString(): string;
+}
+
+export interface NbTokenClass {
+  new (raw: string): NbAuthToken
+}
+
+export function nbCreateToken(tokenClass: NbTokenClass, token: string) {
```

Comment:
Also, I'm not sure this is a good idea to build tokens this way, maybe we can give the user capability provide builder service or just a function, which will be provided instead of token class?

Reply:
yeah, that might make sense if the token builder is at least a bit more complex then the one we have, but it is not and it's really hard to imagine such case.

Reply:
I think token builder can be a simple function such as:
```
token => new MySuperTokenClas(token)
```

It'll be not much harder to provide this simple function. But it'll be much straightforward and easy to understand.